Doc/algo evaluation juin2021

slebastard opened the pull request:
Cette PR enrichit la documentation du modèle, en proposant:
- une page d'évaluation du dernier modèle utilisé, contenant des métriques de machine learning usuelles
- un rapide lexique des métriques de ML
- une explication de notre processus de sélection automatisée des seuils

slebastard committed:
doc: evaluation metrics for model, june 2021

- Adding `evaluation-modele-juin2021.md` which logs evalation metrics, thresholds and volumes of our list categories, generated through https://github.com/signaux-faibles/predictsignauxfaibles/blob/develop/notebooks/03-select-thresholds.ipynb
- Updating doc of algorithme-evaluation with a link to

slebastard committed:
doc: adding link to models benchmark in algorithme-evaluation.md

slebastard committed:
doc: définition termes ML et sélection des seuils

- Ajout d'un lexique poru la définition de quelques termes clés sur
  l'évaluation de notre modèle, notamment pour pouvoir expliquer
  brièvement notre processus de sélection automatique des seuils
- Documentation de la sélection automatique des seuils
- Mise à jour de l'évaluation du modèle de juin 2021 (`evaluation-modele-juin2021.md`)

slebastard committed:
fix: ran `make all` to reformat modified doc

vviers commented on algorithme-evaluation.md:
```diff
@@ -90,9 +93,70 @@ Le modèle est entrainé sur les variables d'apprentissage suivantes:
+Rappel
+: la part d'établissements effectivement positifs étant prédits positifs
+
+Score $F_{\beta}$
```
Latex is unfortunately not supported in GitHub-flavored markdown https://stackoverflow.com/questions/11256433/how-to-show-math-equations-in-general-githubs-markdownnot-githubs-blog

slebastard replied:
Fixed in https://github.com/signaux-faibles/documentation/pull/45/commits/c48bbb45a9145e4bd6bc78ff68bdc5fa308585aa

vviers commented on algorithme-evaluation.md:
```diff
@@ -90,9 +93,70 @@ Le modèle est entrainé sur les variables d'apprentissage suivantes:
+: une métrique d'évaluation prenant à la fois la précision et le rappel en compte, et accordant une importante relative $\beta$ fois plus importante au rappel qu'à la précision
+
+Seuil de classification
+: le modèle Signaux Faibles produit un score de risque entre 0 et 1 produit par notre algorithme. Hors, il faut décider à quelle pallier de risque appartient chaque établissement à partir de ce score de risque. Pour ce faire, et au vu du modèle et de la cible d'apprentissage actuelle, il est nécessaire de définir un premier seuil sur le score de risque au-delà duquel l'établissement est à considé ...
```
```suggestion
: le modèle Signaux Faibles produit un score de risque entre 0 et 1 produit par notre algorithme. Hors, il faut décider à quel pallier de risque appartient chaque établissement à partir de ce score de risque. Pour ce faire, et au vu du modèle et de la cible d'apprentissage actuelle, il est nécessaire de définir un premier seuil sur le score de risque au-delà duquel l'établissement est à considérer "à risque modéré" de défaillance, et un second seuil, plus élevé que le premier, au-delà duquel l'établissement est à considérer "à risque fort" de défaillance. La méthodologie pour déterminer ces seuils est détaillée ci-dessous
```

slebastard replied:
Oups, some spelling mistakes!
Fixed in https://github.com/signaux-faibles/documentation/pull/45/commits/aca623779b339ee2d603892b966e652b5a56d659

vviers commented on algorithme-evaluation.md:
```diff
@@ -90,9 +93,70 @@ Le modèle est entrainé sur les variables d'apprentissage suivantes:
+Plus particulièrement:
+
+- le seuil du pallier "risque fort" est choisi pour maximiser le $F_{0.5}$, une métrique qui favorise deux fois plus la précision que le rappel. Ce score favorise ainsi une précision élevée, et donc l'exclusivité d'établissements effectivement en défaillance dans le pallier "risque fort".
+- le seuil du pallier "risque modéré" est choisi pour maximiser le score $F_2$, qui favorise deux fois plus le rappel que la précision. La maximisation de cette métrique vise à obtenir un palleir "risqe modéré" qui capture un maximum d'établissements effectivement en défaillance, quitte à capturer "par erreur" des faux positifs, c'est-à-dire quitte à viser trop large et lister des entreprises qui ...
```
```suggestion
- le seuil du pallier "risque modéré" est choisi pour maximiser le score $F_2$, qui favorise deux fois plus le rappel que la précision. La maximisation de cette métrique vise à obtenir un pallier "risque modéré" qui capture un maximum d'établissements effectivement en défaillance, quitte à capturer "par erreur" des faux positifs, c'est-à-dire quitte à viser trop large et lister des entreprises qui n'entreront pas en défaillance.
```

slebastard replied:
Also fixed in https://github.com/signaux-faibles/documentation/pull/45/commits/aca623779b339ee2d603892b966e652b5a56d659

vviers commented on evaluation-modele-juin2021.md:
```diff
@@ -0,0 +1,221 @@
+Pallier "risque fort" - $\beta=0.5$ - Optimal threshold: $t_{F1}=0.817$ - $F_{0.5}=0.722$<br>
+Pallier "risque modéré" - $\beta=2$ - Optimal threshold: $t_{F2}=0.179$ - $F_{2}=0.547$
+
+### Evaluation du modèle pré-redressements - seuils $f_{\beta}$
```
Je pense qu'on va très vite oublier d'updater cette partie de la doc, ça me gène un peu

Est-ce qu'on peut imaginer faire pointer cette section vers un JSON dans predictsignauxfaibles qui contient les perfs des modèles ? Qquechose d'un peu plus automatisé ?

vviers replied:
Ah my bad j'avais pas vu que le fichier était `juin2021` spécifiquement. Du coup all good.

Mais je rangerais tous ces fichiers dans un dossier genre `evaluation-modele/juin2021.md` pour ne pas polluer le repo de doc avec le temps

slebastard replied:
Bonne idée!
Fait dans https://github.com/signaux-faibles/documentation/pull/45/commits/9b600f3e198e1eca998807b00166618ec24de33a

vviers commented on evaluation-modele-juin2021.md:
```diff
@@ -0,0 +1,221 @@
+
+### Seuils sélectionnés - $f_{\beta}$
+
+Sur le dataset de test chargé depuis `/home/common/benchmark/052021_full_data_test.csv` (1.2M):<br>
```
je pense que c'est peu être pas top de mettre le chemin vers de la data (même anonymisée) dans notre doc publique (et surtout ça n'apporte pas une information particulièrement utile selon moi)

slebastard replied:
Effectivement, ces chemins ne devraient pas rester là.
Résolu avec https://github.com/signaux-faibles/documentation/pull/45/commits/dd159968ac258a3fd183edfac5bc7241b8bc0ca2

vviers reviewed (approved):
Parfait !

Qques petits commentaires que je te laisse prendre en compte (ou pas ;) ) avant de merger

Par contre, la CI ne passe pas — tu es sur que doctoc a bien tourné et que les changements on bien été committés ?
Plus vicieux, est-ce-que tu as bien la même version de Node / doctoc que la CI  (Node v12)?

slebastard committed:
fix: doctocking from node 12.0.0

slebastard commented:
Running doctoc with node v15.14.0 was indeed causing the CI to fail because of the updated doctoc. Fixed in https://github.com/signaux-faibles/documentation/pull/45/commits/a6e8b6d85bd7a5bbb51e70a6267e8487cc69b832

slebastard committed:
fix: Removing latex from algorithme-evaluation.md

slebastard committed:
fix: Correcting some spelling/grammar mistakes

slebastard committed:
refacto: Putting monthly model evaluation into dedicated folder

slebastard committed:
refacto: evaluation-modele/juin2021.md

- Replacing latex syntax with simple markdown
- Adding datasets volumes
- Removing links to datasets on remote
